README.md
[![Actions Status](https://github.com/genio/alien-sodium/workflows/linux/badge.svg)](https://github.com/genio/alien-sodium/actions)
[![Actions Status](https://github.com/genio/alien-sodium/workflows/macos/badge.svg)](https://github.com/genio/alien-sodium/actions)
[![Actions Status](https://github.com/genio/alien-sodium/workflows/windows/badge.svg)](https://github.com/genio/alien-sodium/actions)

# NAME

Alien::Sodium - Interface to the libsodium library [http://libsodium.org](http://libsodium.org)

# SYNOPSIS

```perl
use strict;
use warnings;

use ExtUtils::MakeMaker;
use Config;
use Alien::Base::Wrapper ();

WriteMakefile(
  Alien::Base::Wrapper->new('Alien::Sodium')->mm_args2(
    ...
    CONFIGURE_REQUIRES => {
      'Alien::Sodium' => '2.000',
    },
    ...
  ),
);
```

# DESCRIPTION

This package can be used by other [CPAN](https://metacpan.org) modules that
require [libsodium](http://libsodium.org).

# SEE ALSO

- [libsodium](http://libsodium.org)

# AUTHOR

Alex J. G. Burzyński <`ajgb@cpan.org`>

# CONTRIBUTORS

Graham Ollis <`plicease@cpan.org`>

# COPYRIGHT & LICENSE

Copyright (c) 2015 Alex J. G. Burzyński. All rights reserved.

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.


Changes
Revision history for Perl extension Alien::Sodium.

{{$NEXT}}

2.000     2020-08-06
  - Bump major version to use decimal versions
  - Upgrade library to version 1.0.18
  - Re-organize Changes file
  - Run CI via GH workflows/actions
  - Keep Makefile.PL and META.json in the repo for easy CI
  - Update documentation
  - Make the README.md mirror the POD
  - Use dzil [@Starter]
  - Use alienfile
  - Add mailmap to get contributors correct
  - Add FFI test
  - Add XS test
  - Keep a license file in the repo
  - Don't package upstream library, download it from their site
  - Get rid of PodWeaver stuff (the docs are small enough)

1.0.8.0  2015-12-27 21:38:06
  - Upgrade library to version 1.0.8

1.0.7.0  2015-12-24
  - Upgrade library to version 1.0.7

1.0.6.0  2015-11-10
  - Upgrade library to version 1.0.6

1.0.3.1  2015-07-16
  - Use correct pkg-config name for libsodium (Graham Ollis - gh#1)
  - Require libsodium 1.0.3 (Graham Ollis - gh#2)

1.0.3.0  2015-05-12
  - Upgrade library to version 1.0.3

0.01     2014-02-04
  - Initial version.


lib/Alien/Sodium.pm
package Alien::Sodium;

use strict;
use warnings;
use utf8;
use parent 'Alien::Base';

our $VERSION = '2.001';

1;

__END__

=encoding utf8

=head1 NAME

Alien::Sodium - Interface to the libsodium library L<http://libsodium.org>

=head1 SYNOPSIS

    use strict;
    use warnings;

    use ExtUtils::MakeMaker;
    use Config;
    use Alien::Base::Wrapper ();

    WriteMakefile(
      Alien::Base::Wrapper->new('Alien::Sodium')->mm_args2(
        ...
        CONFIGURE_REQUIRES => {
          'Alien::Sodium' => '2.000',
        },
        ...
      ),
    );

=head1 DESCRIPTION

This package can be used by other L<CPAN|https://metacpan.org> modules that
require L<libsodium|http://libsodium.org>.

=head1 SEE ALSO

=over 4

=item * L<libsodium|http://libsodium.org>

=back

=head1 AUTHOR

Alex J. G. Burzyński <F<ajgb@cpan.org>>

=head1 CONTRIBUTORS

Graham Ollis <F<plicease@cpan.org>>

=head1 COPYRIGHT & LICENSE

Copyright (c) 2015 Alex J. G. Burzyński. All rights reserved.

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.


=cut


Makefile.PL
# This Makefile.PL for Alien-Sodium was generated by
# Dist::Zilla::Plugin::MakeMaker::Awesome 0.48.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Interface to the libsodium library http://libsodium.org",
  "AUTHOR" => "Alex J. G. Burzy\x{144}ski <ajgb\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Alien::Build" => "2.15",
    "Alien::Build::MM" => "0.32",
    "Alien::Build::Plugin::Build::Autoconf" => "0.04",
    "Alien::Build::Plugin::Build::Make" => 0,
    "Alien::Build::Plugin::Probe::Vcpkg" => "2.15",
    "Config" => 0,
    "ExtUtils::MakeMaker" => "6.52",
    "IPC::Cmd" => 0,
    "Sort::Versions" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "Alien::Build" => "2.15",
    "Alien::Build::MM" => "0.32",
    "Alien::Build::Plugin::Build::Make" => "0.01",
    "Alien::Build::Plugin::Probe::Vcpkg" => "2.15",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "DISTNAME" => "Alien-Sodium",
  "LICENSE" => "perl",
  "NAME" => "Alien::Sodium",
  "PREREQ_PM" => {
    "Alien::Base" => "2.15",
    "base" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.52",
    "File::Spec" => 0,
    "Test2::Suite" => 0,
    "Test2::V0" => 0,
    "Test::Alien" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "2.001",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Alien::Base" => "2.15",
  "Alien::Build" => "2.15",
  "Alien::Build::MM" => "0.32",
  "Alien::Build::Plugin::Build::Autoconf" => "0.04",
  "Alien::Build::Plugin::Build::Make" => 0,
  "Alien::Build::Plugin::Probe::Vcpkg" => "2.15",
  "Config" => 0,
  "ExtUtils::MakeMaker" => "6.52",
  "File::Spec" => 0,
  "IPC::Cmd" => 0,
  "Sort::Versions" => 0,
  "Test2::Suite" => 0,
  "Test2::V0" => 0,
  "Test::Alien" => 0,
  "Test::More" => "0.88",
  "base" => 0,
  "strict" => 0,
  "warnings" => 0
);

# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
use Alien::Build::MM;
my $abmm = Alien::Build::MM->new;
%WriteMakefileArgs = $abmm->mm_args(%WriteMakefileArgs);
# END code inserted by Dist::Zilla::Plugin::AlienBuild

unless ( eval { ExtUtils::MakeMaker->VERSION('6.63_03') } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
{ package
    MY;
  sub postamble {
    $abmm->mm_postamble(@_);
  }
  sub install {
    $abmm->can('mm_install')
      ? $abmm->mm_install(@_)
      : shift->SUPER::install(@_);
  }
}
# END code inserted by Dist::Zilla::Plugin::AlienBuild
